### PR TITLE
Replace println calls with log crate macros

### LINKED ISSUE #42: Log crate instead of println
We're using this broker in a proof of concept and would like to replace the `println!` logs with the `log` crate macros. I can push a PR but want  tot clarify first if this is something mergeable. I would replace and the `println!` calls with `debug!`, `info!`... and initialise `env_logger`(?) in the main.

@bschwind fine for you?

### PR DESCRIPTION
**Pull request by flxo**
Replace the println debug statements with macros from the `log` crate. Add [env-logger](https://crates.io/crates/env_logger) as a dependency and initialize.

Fixes #42

### CODE REVIEW

File: mqtt-v5-broker/src/main.rs [hunk 1]
```diff
@@ -196,6 +199,12 @@ async fn websocket_server_loop(broker_tx: Sender<BrokerMessage>) {
 }
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    if env::var("RUST_LOG").is_err() {
+        env_logger::builder().filter(None, log::LevelFilter::Debug).init();
+    } else {
+        env_logger::init();
+    }
```
Comment (bschwind) [inline mqtt-v5-broker/src/main.rs:206]: Small request, but could you place this within a function named something like `init_logging` or `setup_logging`?

**Review by bschwind — APPROVED** [review]
Looks great, thanks for the PR!

File: mqtt-v5-broker/src/main.rs (hunk 1 above)
Comment (flxo) [reply mqtt-v5-broker/src/main.rs:206]: Sure. Done.

Comment (bschwind) [reply mqtt-v5-broker/src/main.rs:206]: Perfect, thank you!

**Pull request merged**